### PR TITLE
feat: add container whitelisted envs as metrics labels.

### PR DESCRIPTION
**Pull request by Colstuwjx**
Signed-off-by: colstuwjx <colstuwjx@gmail.com>

It's a follow-up PR for #2921 and #2330.

### CODE REVIEW

File: metrics/prometheus.go [hunk 1]
```diff
@@ -1856,6 +1856,9 @@ func BaseContainerLabels(whiteList []string) func(container *info.ContainerInfo)
 				set[ContainerLabelPrefix+k] = v
 			}
 		}
+		for k, v := range container.Spec.Envs {
+			set[ContainerEnvPrefix+k] = v
+		}
```
Comment (iwankgb) [inline metrics/prometheus.go:1861]: A test case would be greatly appreciated.

**Review by iwankgb — CHANGES_REQUESTED** [review]

File: metrics/prometheus.go (hunk 1 above)
Comment (Colstuwjx) [reply metrics/prometheus.go:1861]: Hi @iwankgb, after diving into the code, I found that the env is filtered while manage call container handler, e.g. containerd is at [here](https://github.com/google/cadvisor/blob/aa0afbe/container/containerd/handler.go#L146).

As a result, I added test case `TestPrometheusCollectorWithWhiteList` to cover `BaseContainerLabels` test. If there are any suggestions, pls let me know!

Comment (Colstuwjx) [reply metrics/prometheus.go:1861]: BTW, I found the handler part already did have some test cases about whitelist envs, e.g. see docker [tests](https://github.com/google/cadvisor/blob/aa0afbee6da04e8d8a214333bf2747f0ae27f7a0/container/docker/handler_test.go#L121).

So... I think it's enough.

**Comment by bobbypage** [comment]
LGTM, thanks.

**Pull request merged**